Framework/PythonInterface/plugins/algorithms/SaveISISReflectometryORSO.py

```diff
@@ -138,17 +148,26 @@ def _populate_q_conversion_info(self):
             self._q_conversion_theta = float(np.rad2deg(self._ws.spectrumInfo().signedTwoTheta(0))) / 2.0
 
     def _set_name(self):
+        theta_assigned: bool = self._q_conversion_theta is not None
+
         if self.is_stitched:
             self._name = "Stitched"
-        elif self._q_conversion_theta is not None:
-            self._name = str(self._q_conversion_theta)
+        elif theta_assigned or self.is_polarized:
+            theta_str = f"{self._q_conversion_theta:.3f}" if theta_assigned else ""
+            space_str = " " if (theta_assigned and self.is_polarized) else ""
+            self._name = f"{theta_str}{space_str}{self._spin_state}"
         else:
             self._name = self._ws.name()
 
         # Ensure unique dataset names for workspace group members.
         # Eventually we will specify the polarization spin state to provide the unique names for polarised datasets.
         if self._is_ws_grp_member and self._name != self._ws.name():
-            self._name = f"{self._ws.name()} {self._name}"
+            ws_name = f"{self._ws.name()} " if not self.is_polarized else ""
+            self._name = f"{ws_name}{self._name}"
+
+    def _set_spin_state_from_logs(self) -> None:
+        if self._ws.getRun().hasProperty("spin_state_ORSO"):
```

Comment:
I will look into this. Is there any file where you would prefer the constants to be stored?

Reply:
Sorry, I wrote this comment and then added a similar comment further on because I noticed there were places in the unit tests where we're also needing to make use of constants that we've defined on the C++ side. If you don't mind then I think it would be good to investigate whether it's possible to export the C++ constants to Python. There are a couple of exports that might be useful as reference for this: [DeltaEMode](https://github.com/mantidproject/mantid/blob/main/Framework/PythonInterface/mantid/kernel/src/Exports/DeltaEMode.cpp) and [PhysicalConstants](https://github.com/mantidproject/mantid/blob/main/Framework/PythonInterface/mantid/kernel/src/Exports/PhysicalConstants.cpp), although there may be other more relevant examples that you can find. I'm afraid I've not done this before myself, so I can't guarantee it will turn out to be a feasible option, but I think it's worth investigating as it would be best practice to try not to have the same constants defined in multiple places.

If we find we have to define them separately on the Python side, then I think they should probably go in the ORSO helper file? That way they can be re-used easily across multiple algorithms (and are more likely to be discovered by other developers).